### PR TITLE
Test Squeeze op converter in dynamic shape mode

### PR DESCRIPTION
**Pull request by tfeher**
This PR adds tests for dynamic shape mode for TF-TRT ConvertSqueeze. 

- The BuildAndRun method is extended with a profile generation step.
- ConvertSqueeze is refactored: all test use TestParams
- All previous test in ConvertSqueeze run both in implicit and explicit batch mode
- Added new dynamic shapes tests (those run only in explicit batch mode)

PR #38118 is required for of this PR, please review that first. Tagging @bixia1 for review.

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -2828,125 +2881,180 @@ TEST_F(OpConverterTest, ConvertExpandDims) {
...
-    NodeDef node_def = get_squeeze_nodedef(ok_params[i].axis);
-    AddTestTensor("input", ok_params[i].input_dims);
-    RunValidationAndConversion(node_def);
-    TRT_TensorOrWeights output;
-    TF_EXPECT_OK(GetTensorOrWeights("my_squeeze", &output));
-    ASSERT_TRUE(output.is_tensor());
-    ExpectTrtDimsEqualsArray(ok_params[i].expected_output_dims,
-                             output.tensor()->getDimensions());
+  // Loop trough conversion parameters.
+  for (bool use_implicit_batch : {true, false}) {
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:2936]: Can you use Value-Parameterized Tests instead? Here is an example
https://github.com/tensorflow/tensorflow/blob/958d2c10e663f4bcbbd1035da4201950fc8c96ca/tensorflow/compiler/xla/tests/iota_test.cc#L33-L61
You can also search "value-parameterized tests" and "INSTANTIATE_TEST_CASE_P" for more information.

**Comment by gbaned** [comment]
@tfeher Can you please check @bixia1 comments and keep us posted. Thanks!

**Comment by tfeher** [comment]
Thanks @bixia1 for the review. I am working on fixing the issues, I will update soon.

**Review by tfeher — COMMENTED** [review]
I have addressed the the issues. As part of making the test Value-Parameterized, I have done some additional changes:
- Introduced `TrtTestMode` enum to enumerate test options: implicit/explicit batch & dynamic shape mode.
- Added more test params (apart from TrtTestMode): `Converter` precision and input tensor dtype.

Using these the test coverage of `ConvertSqueeze` is improved: 
- Now all the test run in dynamic shape mode (earlier only those run in dynamic shape mode where we defined partial_input_dims manually). 
- The converter can be tested for multiple input types and conversion precision.

Additional changes:
- Previously, precision mode was set through the `precision_mode_to_test_ ` member variable of `OpConverterTest`. Removed the member variable and added `precision_mode_to_test` as an argument of `OpConverterTest::Reset()`.   `Reset` calls the constructor of the `Converter` and sets the TensorRT precision mode there. 

- Noticed that `precision_mode` arg of `OpConverterTest::BuildAndRun` is not used. Removed it.

To facilitate reusing the same set of changes for other OpConverterTests, I have moved all the inptut tensor initializing  result checking code into helper routines:
- `AddTestTensor` (with `trt_mode` arg), 
- `RunValidationAndConversion`,
- `BuildAndRunConvertedNetwork`

**Review by bixia1 — APPROVED** [review]
Temporarily approving this for myself to further look at this within google.

**Pull request merged**